pkgs/applications/emulators/cdemu/base.nix

```diff
@@ -1,12 +1,16 @@
-{ pname, version, pkgSha256 }:
-attrs@{ lib, stdenv, fetchurl, cmake, pkg-config, buildInputs, drvParams ? {}, ... }:
-stdenv.mkDerivation ( rec {
-  inherit pname version buildInputs;
+{ lib, stdenv, fetchurl, cmake, pkg-config
+, pname, version, sha256, buildInputs
+, nativeBuildInputs ? [ ]
+, postFixup ? ""
+, extraDrvParams ? { }
+}:
+stdenv.mkDerivation ( {
```

Comment:
I would just copy the contents of `base.nix` to each individual  package expression and remove `base.nix`.

It's not like there is much overlap – it really feels like DRY at any cost. But at the end the added piping adds as much duplication as it saves, while being much harder to read and modify (e.g. unable to use `buildPythonApplication` without adding yet another parameter).

Reply:
yeah, I had already considered this. especially if we can use the cmake build hook, I'll go for that.

Reply:
I ended up just renaming `base.nix` to `common-drv-attrs.nix` and inherit into the attribute map from it. That avoids the nested `callPackage`, while still deduplicating source info ...

`buildPythonApplication` can also be used easily with this, but it turns out that this doesn't work right now anyway (or I might be using it wrong, see other comment)